cmd/horcrux/cmd/leader_election.go

```diff
@@ -107,26 +107,40 @@ func getLeaderCmd() *cobra.Command {
 				return fmt.Errorf("threshold mode configuration has no cosigners")
 			}
 
-			keyFile, err := config.KeyFileExistsCosignerRSA()
-			if err != nil {
-				return err
-			}
+			var id int
 
-			key, err := signer.LoadCosignerRSAKey(keyFile)
-			if err != nil {
-				return fmt.Errorf("error reading cosigner key (%s): %w", keyFile, err)
+			keyFile, err := config.KeyFileExistsCosignerECIES()
+			if err == nil {
```

Comment:
We are swallowing the error on line 112. Do we want to log it somewhere? But I see you're trying to determine if there's an ECIES or RSA key, preferring ECIES first.

Checking for `err == nil` is unusual imo. `err != nil` is what Go devs typically expect, although this may be cause for an exception.

Reply:
good call, cleaned this up.